src/compiler/crystal/codegen/debug.cr

```diff
@@ -43,35 +50,46 @@ module Crystal
       # from one module isn't incorrectly used in another module.
       debug_types_per_module =
         @debug_types_per_module ||=
-          {} of LLVM::Module => Hash(Type, LibLLVMExt::Metadata?)
+          {} of LLVM::Module => Hash(String, LibLLVMExt::Metadata?)
```

Comment:
In some cases you have to call underlining type but with the name of the current type as current type is just a wrapper.

Reply:
That was done for this types:
```
    def create_debug_type(type : NilableType, type_name : String? = type.to_s)
      get_debug_type(type.not_nil_type, type_name)
    end

    def create_debug_type(type : NilablePointerType, type_name : String? = type.to_s)
      get_debug_type(type.pointer_type, type_name)
    end
```
if I will pass down type.pointer_type it will create the debug info for that class but not for current type and that information will be incorrect in debug info.